docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,199 @@
+User Customization & Persistence
+================================
+
+Context
+-------
+
+.. todo::
+
+   This section will be handled by the requirements PR.
+
+
+Design Choices
+--------------
+
+:term:`ConfigMap` is chosen as a unified data access and storage media for user
+editable configurations in a MetalK8s cluster based on the above requirements
+for the following reasons:
+
+* Ability to support Update operations on ConfigMap's with CLI and UI easily
+  using our already existing python kubernetes module.
+* Guarantee of adaptability and ease of changing the design and implementation
+  in cases were customer needs evolve rapidly.
+* ConfigMaps stored in the :term:`etcd` database which is generally being
+  backed up. This ensures that user settings cannot be lost easily.
+
+
+How it works
+~~~~~~~~~~~~
+
+During Bootstrap, pillar values for the various services will be pre-filled
+with default values. Service pods and deployments will be configured to either
+consume configuration data directly from their ConfigMap or fallback to
+the default pillar values if the service specific ConfigMap was not found.
+
+**Using Salt states**
+
+Once a ConfigMap is updated by the user (say a user changes the number of
+replicas for Prometheus deployments to a new value), then perform the
+following actions:
+
+  - Apply a Salt states that reads the ConfigMap object, validates the schema
+    and checks the new values passed.
+
+  - Finally, we make sure that the updated values are picked up by their
+    respective consumers (this might require Pod restarts for changes to take
+    effect).
+
+Note that, Salt states are used to sync data and update consumers
+of new configurations changes mainly because of the minimum effort it takes to
+setup this flow but the K8s Operator pattern could be use to replace
+configuration synchronization between user defined configurations and
+consumers.
+
+Non-goals
+~~~~~~~~~
+
+- Persisting newly added Grafana dashboards or new Grafana datasources
+  especially for modifications added via the Grafana UI cannot be stored in
+  ConfigMaps.
+  To handle this particular user case, we need to configure Grafana deployments
+  to use persistent volume claims and then document how to create volumes that
+  will handle the persistence of these settings across Pod restarts.
+
+- ConfigMaps as used in the above design for storing and persisting user
+  settings need periodic backups for recovery purposes. This implies that
```

Comment:
I get what you mean but I don't like how it's phrased. We don't need to put backups in place to store these user settings and such. Instead, using `ConfigMap`s (or other K8s object types) has as a benefit that K8s cluster backups include these settings. The fact that the K8s `etcd` cluster needs to be backed up is not driven by this feature.